composer/src/lib.rs

```diff
@@ -410,6 +458,24 @@ impl FromIterator<ComposerPackage> for ComposerRepository {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum ComposerRepositoryDisablement {
+    Boolean(MustBe!(false)), // a plain false value
+    #[allow(clippy::zero_sized_map_values)]
+    Object(HashMap<String, MustBe!(false)>), // a (single-field) object like {"packagist.org": false}
+}
+
+impl FromStr for ComposerRepositoryDisablement {
+    type Err = String;
```

Comment:
We can never fail turning a &str into a `String`, we can make that clearer with a different error type 

```suggestion
    type Err = Infallible;
```

Reply:
Ah yes good call. In fact, this should be a `From<&str>` and not `FromStr`, even simpler!

Reply:
Updated accordingly.